Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java

```diff
@@ -87,6 +91,21 @@ protected void onCreate(Bundle savedInstanceState) {
 
         setTitle(getResources().getString(R.string.text_shopping));
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
too much code repitition.  Can there be a simpler way for this?

Reply:
I _could_ create a public method that takes the permission of a parameter and requests it if not already, and call that in the required places.

Reply:
Let's have a new interface to request for specific permissions and our class can implement whichever needed. "Interface segregation principle" can be applied here. Let me know if you need any help here.

Reply:
What do you mean by interface? Should I create methods that a class will only implement if required?

Reply:
@flash76 Interface is a blueprint for a class, you can declare methods in interfaces and let the class implementation to it implement the actual method.
You can read about interfaces here: https://www.geeksforgeeks.org/interfaces-in-java/